ethereum-client/src/lib.rs

```diff
@@ -243,15 +245,19 @@ pub fn get_block(
     })
 }
 
-pub fn get_block_object(server: &str, block_num: &str) -> Result<BlockObject, EthereumClientError> {
+pub fn get_block_object(
+    server: &str,
+    block_num: &str,
+    deadline: u64,
+) -> Result<BlockObject, EthereumClientError> {
     let params = vec![block_num.into(), true.into()];
-    let response_str: String = send_rpc(server, "eth_getBlockByNumber".into(), params)?;
+    let response_str: String = send_rpc(server, "eth_getBlockByNumber".into(), params, deadline)?;
     let response = deserialize_get_block_by_number_response(&response_str)?;
     response.result.ok_or(EthereumClientError::JsonParseError)
 }
 
-pub fn get_latest_block_number(server: &str) -> Result<u64, EthereumClientError> {
-    let response_str: String = send_rpc(server, "eth_blockNumber".into(), vec![])?;
+pub fn get_latest_block_number(server: &str, deadline: u64) -> Result<u64, EthereumClientError> {
```

Comment:
Oh I don't think we want to have to pass this in?

Reply:
@jflatow do you mean that the deadline should be read from runtime_interfaces inside ethereum-client?

Reply:
I think it should just be a compile time `const`?

Reply:
Ok, so the overthinking here, I will move it to just a const var